File: easyserver/src/main/java/org/ecjtu/easyserver/server/util/hash/HashUtil.java
package org.ecjtu.easyserver.server.util.hash;

/**
 * Created by KerriGan on 2017/8/20.
 */

public class HashUtil {
    public static Long BKDRHash(String str){
        int seed=131;
        Long hash=0L;
        int len=str.length();
        int index=0;
        while (index < len){
            hash = hash * seed + (long)(str.charAt(index++));
        }
        return hash & 0x7FFFFFFF;
    }
}

File: app/src/main/java/com/ecjtu/sharebox/util/hash/HashUtil.kt
package com.ecjtu.sharebox.util.hash

/**
 * Created by Ethan_Xiang on 2017/8/14.
 */
object HashUtil {
    fun BKDRHash(string: String): Long {
        val seed = 131
        var hash = 0L
        val len = string.length
        var index = 0
        while (index < len) {
            hash = hash * seed + (string[index++].toInt())
        }
        return (hash and 0x7FFFFFFF)
    }
}